lib/ansible/modules/database/mongodb/mongodb_role.py

```diff
@@ -0,0 +1,452 @@
+
+# Create role 'producer' in database 'test' with 'insert' privilege to 'whatever' collection in database 'test' and 'readWrite' access in database 'private'.
+- mongodb_role:
+    database: test
+    name: producer
+    roles:
+        - db: private
+          role: readWrite
+    privileges:
+        - resource:
+            db: test
+            collection: whatever
+          actions:
+            - insert
+    state: present
+
+# Remove role 'kaktus' from database 'test' via SSL (MongoDB must be compiled with the SSL option and configured properly)
+- mongodb_role:
+    database: test
+    name: kaktus
+    state: absent
+    ssl: True
+'''
+
+RETURN = '''
+role:
+    description: The name of the role to add or remove.
+    returned: success
+    type: string
+'''
+
+import os
+import ssl as ssl_lib
+import traceback
+from distutils.version import LooseVersion
+
+try:
+    import yaml
+    HAS_YAML = True
+except ImportError:
+    HAS_YAML = False
+
+try:
+    from pymongo.errors import ConnectionFailure
+    from pymongo.errors import OperationFailure
+    from pymongo import version as PyMongoVersion
+    from pymongo import MongoClient
+except ImportError:
+    try:  # for older PyMongo 2.2
+        from pymongo import Connection as MongoClient
+    except ImportError:
+        HAS_PYMONGO = False
+    else:
+        HAS_PYMONGO = True
+else:
+    HAS_PYMONGO = True
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.six.moves import configparser
+from ansible.module_utils._text import to_native
+
+
+# =========================================
+# MongoDB module specific support methods.
+#
+
+def check_compatibility(module, client):
+    """Check the compatibility between the driver and the database.
+
+       See: https://docs.mongodb.com/ecosystem/drivers/driver-compatibility-reference/#python-driver-compatibility
+
+    Args:
+        module: Ansible module.
+        client (cursor): Mongodb cursor on admin database.
+    """
+    loose_srv_version = LooseVersion(client.server_info()['version'])
+    loose_driver_version = LooseVersion(PyMongoVersion)
+
+    if loose_srv_version >= LooseVersion('3.2') and loose_driver_version < LooseVersion('3.2'):
+        module.fail_json(msg=' (Note: you must use pymongo 3.2+ with MongoDB >= 3.2)')
+
+    elif loose_srv_version >= LooseVersion('3.0') and loose_driver_version <= LooseVersion('2.8'):
+        module.fail_json(msg=' (Note: you must use pymongo 2.8+ with MongoDB 3.0)')
+
+    elif loose_srv_version >= LooseVersion('2.6') and loose_driver_version <= LooseVersion('2.7'):
+        module.fail_json(msg=' (Note: you must use pymongo 2.7+ with MongoDB 2.6)')
+
+    elif LooseVersion(PyMongoVersion) <= LooseVersion('2.5'):
+        module.fail_json(msg=' (Note: you must be on mongodb 2.4+ and pymongo 2.5+ to use the roles param)')
+
+
+def load_mongocnf(config_file):
+    config = configparser.RawConfigParser()
+
+    try:
+        with open(config_file) as f:
```

Comment:
wrap these pieces of code into two try/except statements to be precise with what you're trapping.
1) one around with-block (with readfp call) catching `IOError`
2) the other one around creds dict constuction catching `configparser.NoOptionError`